api/authenticators/okta/okta_auth.go

```diff
@@ -305,20 +304,82 @@ func (o *oktaAuthClient) GetSAMLResponse(ctx context.Context, application okta.A
 	// This response will redirect us to the SAML Endpoint
 	resp, err = client.Do(req)
 	if err != nil {
-		return nil, err
+		return nil, TranslateError(resp, err)
 	}
 
 	// Now we have the SAML URL, we can send a request to that URL with our cookie to get the SAML response
 	req, err = http.NewRequest("GET", resp.Header.Get("Location"), nil)
 	if err != nil {
-		return nil, err
+		return nil, WrapError(err, ErrOktaSAMLError)
 	}
 
 	req.AddCookie(sid)
 	resp, err = client.Do(req)
 	if err != nil {
-		return nil, err
+		return nil, TranslateError(resp, err)
+	}
+
+	samlResponse, err := extractSAMLResponse(resp.Body)
+	if err != nil {
+		return samlResponse, WrapError(err, ErrOktaSAMLError)
+	}
+
+	return samlResponse, nil
+}
+
+// OktaError is an error from Okta.
+type OktaError error
+
+// A list of standard errors that can be returned by by Okta client.
+var (
+	ErrOktaBadRequest            OktaError = errors.New("bad request")
+	ErrOktaUnauthorized          OktaError = errors.New("unauthorized")
+	ErrOktaForbidden             OktaError = errors.New("forbidden")
+	ErrOktaCouldNotCreateSession OktaError = errors.New("could not create a session")
+	ErrOktaSAMLError             OktaError = errors.New("could not get a SAML response")
+	ErrOktaInternalServerError   OktaError = errors.New("internal server error")
+	ErrOktaUnspecified           OktaError = errors.New("unspecified")
+)
+
+// WrapError wraps an error into a standard Okta client error.
+func WrapError(err error, oktaErr OktaError) error {
```

Comment:
I don't think this needs to be exported. You should rename this to `wrapError` so it is not visible outside of this package.